includes/class-wcs-post-meta-cache-manager.php

```diff
@@ -248,12 +248,12 @@ protected function trigger_delete_all_caches_hook( $meta_key ) {
 	}
 
 	/**
-	 * Abstract the check against get_post_type() so that it can be mocked for unit tests.
+	 * Abstract the check against get_order_type() so that it can be mocked for unit tests.
 	 *
-	 * @param int $post_id Post ID or post object.
-	 * @return bool Whether the post type for the given post ID is the post type this instance manages.
+	 * @param int $id Subscription ID or subscription object itself.
+	 * @return bool Whether the order type for the given ID is the order type this instance manages.
 	 */
-	protected function is_managed_post_type( $post_id ) {
-		return $this->post_type === get_post_type( $post_id );
+	protected function is_managed_post_type( $id ) {
+		return WC_Data_Store::load( 'subscription' )->get_order_type( $id ) === $this->post_type;
```

Comment:
After thinking about this change a little more, I don't think we need to update this function because the `Post_Meta_Cache_Manager` class is predicated on the assumption that this class will be used for any posts.

Therefore loading the subscriptions data store and checking order type doesn't make sense here and might add confusion (even though the CPT `get_order_type()` is just a wrapper for `get_post_type()`.

For stores that have HPOS is enabled, we'll now use James' new `WCS_Object_Data_Cache_Manager` class being added in #304.

So I say let's leave this function using `$this->post_type === get_post_type( $post_id );` and I'll make a comment on James' PR to add a note to this class to specify it's only used for WP Posts and also make sure we're overriding `is_managed_post_type()` in the new `WCS_Object_Data_Cache_Manager` class (which extends this)

Let me know if that makes sense.

Reply:
It makes sense. Reverted in [2b1d9d5](https://github.com/Automattic/woocommerce-subscriptions-core/pull/318/commits/2b1d9d5cc44188df80af84a62d25c762fa681ab7).